src/algorithms/util/matrixToListConverter.java

```diff
@@ -0,0 +1,35 @@
+package src.algorithms.util;
+import java.util.*;
+/**
+ * Supports method to convert an Adjacency Matrix into an Adjacency list
+ * Adjacency Matrix in the form of a 2D Integer Array int[][]
+ * Adjacency list in the form of a 2D Integer array int[][], where array[i] is a list of neighbour indexes
+ */
+
+public class matrixToListConverter {
+
+    public int[][] convert(int[][] adjM) {
+
+        List<ArrayList<Integer>> adjL = new ArrayList<>();
+        int numOfNodes = adjM.length;
+
+        for (int i = 0; i < numOfNodes; i++) {
+            adjL.add(new ArrayList<>());
+            for (int j = 0; j < numOfNodes; j++) {
+                if (adjM[i][j] == 1) {
+                    adjL.get(i).add(j);
+                }
+            }
+        }
+
+        int[][] res = new int[numOfNodes][];
+        for (int i = 0; i < numOfNodes; i++) {
+            ArrayList<Integer> neighbourList = adjL.get(i);
+            for (int j = 0; j < neighbourList.size(); j++) {
+                res[i][j] = neighbourList.get(j);
+            }
+        }
+
+        return res;
+    }
```

Comment:
[optional] Since this is a sub-routine that will be called, in many places, would be nice to create a 'pretty' function to help visualise and possibly debug. 
eg:

Adjacency list of vertex 0
 head -> 1-> 4

 Adjacency list of vertex 1
 head -> 0-> 2-> 3-> 4

 Adjacency list of vertex 2
 head -> 1-> 3

 Adjacency list of vertex 3
 head -> 1-> 2-> 4

 Adjacency list of vertex 4
 head -> 0-> 1-> 3

i randomly googled and saw GFG has the above [here](https://www.geeksforgeeks.org/graph-and-its-representations/). Feel free to do anything else too.